cmd/snap/cmd_run.go

```diff
@@ -748,6 +752,61 @@ func activateXdgDocumentPortal(info *snap.Info, snapApp, hook string) error {
 
 type envForExecFunc func(extra map[string]string) []string
 
+var gdbServerWelcomeFmt = `
+Welcome to "snap run --gdbserver".
+You are right before your application is run.
+Please open a different terminal and run:
+
+gdb -ex="target remote %[1]s" -ex="continue" -ex="signal SIGCONT"
```

Comment:
A bunch of SIGSTOPs flying around from gdbserver and the process itself. What works for me and stops at SIGTRAP is:
```suggestion
gdb -ex="target remote %[1]s" -ex=continue -ex="signal SIGCONT" -ex=continue
```